### DISCUSSION TITLE
ryot and OIDC

### DISCUSSION THREAD

**Discussion by bunkerman** [IgnisDa/ryot#878] [Q&A] [answered]
Hi, I started using ryot and having a problem a startup. I'm using authelia for SSO and set "USERS_DISABLE_LOCAL_AUTH=true". I looks like ryot is started before authelia comes up. So ryot tries to connect to Authelia, but fails at startup. I just see a blank page where the OIDC button is missing. I have to manually restart the container. Is it possible that ryot checks in intervals for the OIDC connection or can I delay the start of the ryot container in some way?

**Comment by IgnisDa** [comment]
You can use the `depends` settings of docker compose to deplay Ryot startup. https://docs.docker.com/compose/compose-file/05-services/#depends_on

**Reply by bunkerman** [reply]
Thanks for the tip, but that unfortunately doesn't work in my environment. I keep the compose file separated by container/usecase, so my authelia compose file is in another folder than ryot. Is it possible to add checks to ryot until it has a positive connection to the OIDC server?

**Reply by IgnisDa** [reply]
Not sure I want to implement that since the connection is made just once and then re-used across the application. Doing it in the way you described will result in increased memory usage.

Help me understand: why can't you `up` the Authelia container first and then the Ryot stack?

**Reply by bunkerman** [reply]
I shutdown my server/NAS over night and it wakes at a certain time. All my containers start at once so I can't control which container comes up first, unless they share a compose file. I mainly try to keep certain (integral) containers like pihole and authelila separated in the compose files, because I manage them in portainer and it's easier for me to overlook them in that way. Hope that makes sense.

**Comment by IgnisDa** [comment]
It might be possible to integrate [wait-for-it](https://github.com/vishnubob/wait-for-it) in Ryot but I am not sure how it would handle it gracefully.

**Reply by bunkerman** [reply]
I also found this, but don't know if that works on ryot: https://forum.syncthing.net/t/docker-compose-add-startup-delay-to-entrypoint/21490

**Reply by IgnisDa** [reply]
How much time does authelia take to startup? I can add a `sleep ${SOME_NUM_FROM_ENV_VARIABLE}` to the entrypoint.

**Comment by IgnisDa** [comment, marked as answer]
Released `v6.2.5` with `SERVER_SLEEP_BEFORE_STARTUP_SECONDS` env variable (defaults to 0). You can use that in combination with `/health` endpoint to check service healthiness.

**Reply by bunkerman** [reply]
Thank you so much for taking the time. I will try it!
